package a.c.a.b;

import a.c.a.b.b;
import java.util.HashMap;

/* loaded from: classes.dex */
public class a<K, V> extends b<K, V> {
    public HashMap<K, b.c<K, V>> f = new HashMap<>();

    @Override // a.c.a.b.b
    public b.c<K, V> a(K k) {
        return this.f.get(k);
    }

    public boolean contains(K k) {
        return this.f.containsKey(k);
    }

    @Override // a.c.a.b.b
    public V d(K k) {
        V v = (V) super.d(k);
        this.f.remove(k);
        return v;
    }

    public V e(K k, V v) {
        b.c<K, V> cVar = this.f.get(k);
        if (cVar != null) {
            return cVar.f372c;
        }
        this.f.put(k, c(k, v));
        return null;
    }
}
